File: modules/0104_exploratory_data_analysis/03_data_management_in_R/script_03.R

salary_data <- read.csv("../../../data/0104_exploratory_data_analysis/data_management/sal_data.csv", header = TRUE)
head(salary_data)

bonus_data <- read.csv("../../../data/0104_exploratory_data_analysis/data_management/bonus_data.csv", header = TRUE)
head(bonus_data)


left <- merge(salary_data, bonus_data, by = c("Employee_ID"), all.x = TRUE)
head(left)

right <- merge(salary_data, bonus_data, by = c("Employee_ID"), all.y = TRUE)
head(right)

inner <- merge(salary_data, bonus_data, by = c("Employee_ID"))
head(inner)

outer <- merge(salary_data, bonus_data, by = c("Employee_ID"), all = TRUE)
head(outer)


salary_01 <- read.csv("../../../data/0104_exploratory_data_analysis/data_management/basic_salary - 1.csv", header = TRUE)
salary_02 <- read.csv("../../../data/0104_exploratory_data_analysis/data_management/basic_salary - 2.csv", header = TRUE)
salary_rb <- rbind(salary_01, salary_02)
salary_rb


a1 <- aggregate(ms ~ Location, data = salary_data, FUN = sum)
a1

a2 <- aggregate(ms ~ Location + Grade, data = salary_data, FUN = sum)
a2

a3 <- aggregate(cbind(ba, ms) ~ Location, data = salary_data, FUN = sum)
a3

a4 <- aggregate(cbind(ba, ms) ~ Location + Grade, data = salary_data, FUN = sum)
a4


gb1 <- group_by(salary_data, Location)
ag1 <- summarize(gb1, Total = sum(ms, na.rm = TRUE))
ag1

File: scripts/0104_exploratory_data_analysis/03_data_management/script_03.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Oct  2 16:39:01 2023

@author: jerry
"""



# %%

import pandas as pd



# %%

salary_data = pd.read_csv("./data/0104_exploratory_data_analysis/03_data_management/sal_data.csv")
bonus_data = pd.read_csv("./data/0104_exploratory_data_analysis/03_data_management/bonus_data.csv")



# %%

left_join = pd.merge(salary_data, bonus_data, how = 'left')
left_join



# %%

right_join = pd.merge(salary_data, bonus_data, how = 'right')
right_join



# %%

inner_join = pd.merge(salary_data, bonus_data)
inner_join



# %%

outer_join = pd.merge(salary_data, bonus_data, how = 'outer')
outer_join



# %%

salary_1 = pd.read_csv("./data/data_management/basic_salary - 1.csv")
salary_2 = pd.read_csv("./data/data_management/basic_salary - 2.csv")

pd.concat([salary_1, salary_2])



# %%

salary_data = pd.read_csv("./data/data_management/basic_salary.csv")

salary_data.groupby('Location')['ms'].sum()
salary_data.groupby('Location')['ba', 'ms'].sum()
salary_data.groupby(['Location', 'Grade'])[['ba', 'ms']].sum()
